### PR TITLE
Update compile test to use ui_test 0.23

### LINKED ISSUE #13241: Update `ui_test` to 0.23.0
[`ui_test`](https://github.com/oli-obk/ui_test) has updated to 0.23.0, fixing https://github.com/oli-obk/ui_test/issues/211. This means that the version number can be removed for the dependency on all compile fail tests.

https://github.com/bevyengine/bevy/blob/77ebabc4fe0224565a2039bd9c0195901560b67f/crates/bevy_ecs/compile_fail/Cargo.toml#L10-L12

This is a non-trivial update with breaking changes, so it will require a bit more effort than changing a version number.

cc @Brezak

### PR DESCRIPTION
**Pull request by Brezak**
# Objective

Closes #13241

## Solution

Update test utils to use `ui_test` 0.23.0.

## Testing

- Run compile tests for bevy_ecs.

cc @BD103

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ exclude = [
   "crates/bevy_derive/compile_fail",
   "crates/bevy_ecs/compile_fail",
   "crates/bevy_reflect/compile_fail",
+  "tools/compile_fail_utils"
```
Comment (BD103) [inline Cargo.toml:21]: Good catch! I forgot to cover this in #13196 :)

File: tools/compile_fail_utils/src/lib.rs [hunk 2]
```diff
@@ -45,6 +52,11 @@ fn basic_config(root_dir: impl Into<PathBuf>, args: &Args) -> Config {
         "$HOME",
     );
 
+    config.comment_defaults.base().custom.insert(
```
Comment (BD103) [inline tools/compile_fail_utils/src/lib.rs:55]: (Non-blocking) Could you add a comment explaining what this line is doing? It's a bit confusing.

**Review by BD103 — APPROVED** [review]
Tested locally and it looks good, thanks for the quick work! I only have one nit about a confusing line of code (I'm assuming it is the replacement for `dependencies_crate_manifest_path`).

**Review by mockersf — APPROVED** [review]
ok for me with the comment on `config.comment_defaults`

**Review by BD103 — APPROVED** [review]
Please fix CI, and you're good to go!

**Pull request merged**